[css-transitions] Fix transition tests (part 2)

birtles opened the pull request:
This is the first batch of fixes to these tests. I also have fixes for `detached-container-001.html` and `hidden-container-001.html` that have landed on mozilla-central and should eventually be synced back to this repository before long.

birtles committed:
[css-transitions] Simplify before-DOMContentLoaded-001.html test

This test is still not awesome. It introduces a 3s delay which will be
too long most of the time but probably not long enough in some extra
cases (very loaded down CI etc.).

Unfortunately, there doesn't appear to be any way to arbitrarily delay
DOMContentLoaded in a way that can be aborted (short of pehaps setting
up a server script that we communicate with out-of-band). Basically,
only script loading delays DOMContentLoaded (external stylesheets block
rendering, but not DOMContentLoaded) yet there doesn't appear to be any
facility in HTML for aborting a script load (dropping the <script>
element, replacing its 'src' attribute etc. etc. don't work).

So this is probably as good as we can do.

I'm actually going to drop this test in the next commit in this series
but I thought I'd land these fixups for now for posterity.

birtles committed:
[css-transitions] Drop before-DOMContentLoaded-001.html

Dropping this test because:

* It tests unspecified behavior (as indicated in README)
* It fails in at least Firefox and Chrome (and probably every UA since
  it refers to a resource that, as best I can tell, does not exist)
* There doesn't appear to be any way to make it run in a manner that is
  both performant and reliable (see previous commit)

birtles committed:
[css-transitions] Fix before-load-001.html

birtles committed:
[css-transitions] Rewrite tests for testing changes to transition properties while transitions are in-flight

birtles committed:
[css-transitions] Tidy up currentcolor test

This test is quite old and has been updated to test the opposite of what
it was originally intended to test.[1] As a result the (broken) link to
the W3C bugzilla instance[2] only adds confusion.

This patch drops some old links, updates others, and generally tries to
make the test a little easier to follow.

[1] https://github.com/web-platform-tests/wpt/commit/09e605708ee23db287451c0f379374e3b5f3f2eb
[2] Try this link instead: https://web.archive.org/web/20170716073923/https://www.w3.org/Bugs/Public/show_bug.cgi?id=14605

graouts commented on css/css-transitions/before-load-001.html:
```diff
@@ -1,150 +1,46 @@
+<meta name="assert" content="Test checks that transitions are run even before the load event fires">
+<!-- NOTE: This test covers unspecified behavior and should probably be
+     removed. It this behavior *were* specified, it would probably be
+     specified here: -->
```
Is this note still current?

birtles replied:
Yeah, I don't think anything requires this to happen. That said, it passed on the browsers I tested it on. I'm not sure if it makes sense to specify that or not though so perhaps we should just drop the test/

graouts commented on css/css-transitions/before-load-001.html:
```diff
@@ -1,150 +1,46 @@
+  getComputedStyle(div).height;
+  div.style.height = '100px';
+
+  div.addEventListener('transitionend', t.step_func_done(() => {
```
Would it not be sufficient to listen to `transitionrun` here? This should be indicative of a transition being able to run.

birtles replied:
Yeah, I think initially I tried to avoid using transitionrun since it wasn't so widely supported. Later I ended up using it in other tests though so I should probably update this like you suggest.

graouts commented on css/css-transitions/changing-while-transition-002.html:
```diff
@@ -0,0 +1,56 @@
+
+</head>
+<body>
+<div id=log></div>
```
Consistency, and shall I say taste, would dictate double quotes around `log`. This is true of a other tests as well.
```suggestion
<div id="log"></div>
```

birtles replied:
Fair enough. I've started dropping unnecessary quotes more and more. I guess my taste is getting worse :)

graouts replied:
This was said (mostly) in jest, this is quite a subjective topic. But I think we ought to be consistent and use quotes around attributes like in other places.

graouts reviewed (approved):
Very nice, just a few minor comments.

birtles committed:
[css-transitions] Address review feedback

birtles commented:
I've addressed the review feedback except for the question about `before-load-001.html`. I'm not sure if we should keep that or not but I start to suspect we should not.

birtles commented:
FWIW, the other incoming changes for these tests are here:
* https://github.com/web-platform-tests/wpt/pull/16518
* https://github.com/web-platform-tests/wpt/pull/16523

birtles commented:
I've included `before-load-001.html` for now. It's easy to drop later.
